src/main/java/io/pravega/connectors/flink/PravegaInputSplit.java

```diff
@@ -66,22 +64,30 @@ public boolean equals(Object o) {
             return false;
         }
 
-        String thisScope = segmentRange.getScope();
-        String thatScope = that.getSegmentRange().getScope();
-        if (thisScope == null ? thatScope != null : !thisScope.equals(thatScope)) {
-            return false;
-        }
-
-        String thisStream = segmentRange.getStreamName();
-        String thatStream = that.getSegmentRange().getStreamName();
-        if (thisStream == null ? thatStream != null : !thisStream.equals(thatStream)) {
-            return false;
+        if (segmentRange != null) {
```

Comment:
do we have unit test that validates the equals() method for various test conditions? If not, could you please add?